packages/zoe/src/contractSupport/priceAuthority.js

```diff
@@ -236,26 +237,57 @@ export function makeOnewayPriceAuthorityKit(opts) {
       assertBrands(brandIn, brandOut);
       return timer;
     },
-    getPriceNotifier(brandIn, brandOut) {
-      assertBrands(brandIn, brandOut);
-      return notifier;
+    makeQuoteNotifier(amountIn, brandOut) {
+      amountMath.coerce(amountIn, actualBrandIn);
+      assertBrands(amountIn.brand, brandOut);
+
+      // Wrap our underlying notifier with specific quotes.
+      const specificBaseNotifier = harden({
+        async getUpdateSince(updateCount = NaN) {
+          // We use the same updateCount as our underlying notifier.
+          const record = await E(notifier).getUpdateSince(updateCount);
+
+          // We create a quote inline.
+          const quote = createQuote(calcAmountOut => ({
+            amountIn,
+            amountOut: calcAmountOut(amountIn),
+          }));
+          assert(quote);
+
+          const value = await quote;
```

Comment:
Would it make sense to wait just once for the two awaits here?

Reply:
We have to wait for the `updateCount` update to resolve before we can create a quote (since the price may have changed).